hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/index/RunIndexActionExecutor.java

```diff
@@ -167,6 +167,10 @@ public Option<HoodieIndexCommitMetadata> execute() {
       } else {
         String indexUptoInstant = fileIndexPartitionInfo.getIndexUptoInstant();
         // save index commit metadata and update table config
+        // instantiation of metadata writer will automatically instantiate the partitions.
+        table.getIndexingMetadataWriter(instantTime)
```

Comment:
this is what i was talking about.. we need to trigger the mdt instantiation in this block too.